react/src/components/forms/InputCurrency/index.js

```diff
@@ -140,15 +140,15 @@ const Currency = (props) => {
           const handleBlur = (e) => {
             const { type } = e;
             const inputEl = ref.current;
-            const stringValue = inputEl.value;
-            // isNotNumber returns true if stringValue is null, undefined or 'NaN'
+            const { value } = inputEl;
+            const numberValue = Number(numbro.unformat(value));
+            // isNotNumber returns true if value is null, undefined or NaN vs Number.isNaN only checks if value is NaN
             /* eslint-disable-next-line   no-restricted-globals */
-            const isNotNumber = !stringValue || isNaN(Number(numbro.unformat(stringValue)));
+            const isNotNumber = isNaN(value);
             if (isNotNumber) {
               inputEl.setAttribute('placeholder', props.placeholder);
-            }
-            let newValue = isNotNumber ? '' : Number(numbro.unformat(stringValue));
-            if (!is.empty(newValue)) {
+            } else {
+              let newValue = isNotNumber || numberValue;
```

Comment:
In the else, we already know `isNotNumber` is false, so this should be:
```suggestion
              let newValue = numberValue;
```